client/src/test/java/org/ehrbase/client/openehrclient/defaultrestclient/DefaultRestAqlEndpointTestIT.java

```diff
@@ -158,6 +162,205 @@ public void testExecuteEntityQuery() {
+                    containmentComposition.EHRBASE_BLOOD_PRESSURE_SIMPLE_DE_V0_COMPOSITION
+            );
+            Parameter<UUID> ehrIdParameter = entityQuery.buildParameter();
+
+            Condition where = Condition.equal(EhrFields.EHR_ID(), ehrIdParameter);
+            entityQuery.where(where).orderBy(t.orderBy);
+
+            assertThat(openEhrClient.aqlEndpoint().execute(entityQuery, ehrIdParameter.setValue(ehr)))
+                    .extracting(Record1::value1)
+                    .extracting(EhrbaseBloodPressureSimpleDeV0Composition::getVersionUid)
+                    .extracting(VersionUid::getUuid)
+                    .as("TestCase %s", t.id)
+                    .containsExactly(t.uuids);
+
+        });
+    }
+
+
+    @Test
+    public void testExecuteEntityTOP() {
+
+        UUID ehr = openEhrClient.ehrEndpoint().createEhr();
+
+        EhrbaseBloodPressureSimpleDeV0Composition comp1 = openEhrClient.compositionEndpoint(ehr).mergeCompositionEntity(TestData.buildEhrbaseBloodPressureSimpleDeV0());
+
+        EhrbaseBloodPressureSimpleDeV0Composition ehrbaseBloodPressureSimpleDeV0Composition = TestData.buildEhrbaseBloodPressureSimpleDeV0();
+        ehrbaseBloodPressureSimpleDeV0Composition.getBloodPressureTrainingSample().get(0).setSystolicMagnitude(44d);
+        EhrbaseBloodPressureSimpleDeV0Composition comp2 = openEhrClient.compositionEndpoint(ehr).mergeCompositionEntity(ehrbaseBloodPressureSimpleDeV0Composition);
+
+
+        EhrbaseBloodPressureSimpleDeV0Composition ehrbaseBloodPressureSimpleDeV0Composition2 = TestData.buildEhrbaseBloodPressureSimpleDeV0();
+        ehrbaseBloodPressureSimpleDeV0Composition2.getBloodPressureTrainingSample().get(0).setSystolicMagnitude(44d);
+        ehrbaseBloodPressureSimpleDeV0Composition2.getBloodPressureTrainingSample().get(0).setDiastolicMagnitude(44d);
+        EhrbaseBloodPressureSimpleDeV0Composition comp3 = openEhrClient.compositionEndpoint(ehr).mergeCompositionEntity(ehrbaseBloodPressureSimpleDeV0Composition2);
+
+        EhrbaseBloodPressureSimpleDeV0CompositionContainment containmentComposition = EhrbaseBloodPressureSimpleDeV0CompositionContainment.getInstance();
+
+        BloodPressureTrainingSampleObservationContainment containmentObservation = BloodPressureTrainingSampleObservationContainment.getInstance();
+
+        containmentComposition.setContains(containmentObservation);
+
+        class TestCase {
+            int id;
+            TopExpresion topExpresion;
+            UUID[] uuids;
+
+            TestCase(int id, TopExpresion topExpresion, UUID... uuids) {
+                this.id = id;
+                this.topExpresion = topExpresion;
+                this.uuids = uuids;
+            }
+        }
+
+        List<TestCase> testCases = new ArrayList<>();
+
+
+        testCases.add(new TestCase(1,
+                TopExpresion.forward(1),
+                comp3.getVersionUid().getUuid()));
+
+        /* Direction  is ignored on ehrbase
```

Comment:
So this is a ToDo?

Reply:
yes see https://github.com/ehrbase/ehrbase/issues/265